app/services/parameters/caster.rb

```diff
@@ -0,0 +1,118 @@
+module Parameters
+  class Caster
+    TRUE_VALUES = [true, 1, '1', 't', 'T', 'true', 'TRUE', 'on', 'ON', 'yes', 'YES', 'y', 'Y'].to_set
+    FALSE_VALUES = [false, 0, '0', 'f', 'F', 'false', 'FALSE', 'off', 'OFF', 'no', 'NO', 'n', 'N'].to_set
+
+    def initialize(item, options = {})
+      defaults = {
+        :attribute_name => :value,
+        :to => :string
+      }
+      options.reverse_merge!(defaults)
+      @item, @options = item, options
+    end
+
+    def cast!
+      @item.send("#{@options[:attribute_name]}=", casted_value)
+    end
+
+    private
+
+    def casted_value
+      case @options[:to]
+      when :string
+        cast_string
+      when :integer
+        cast_integer
+      when :real
+        cast_real
+      when :boolean
+        cast_boolean
+      when :array
+        cast_array
+      when :hash
+        cast_hash
+      when :json
+        cast_json
+      when :yaml
+        cast_yaml
+      else
+        #cowardly refusing to cast
+        value
+      end
+    end
+
+    def cast_string
+      value.to_s
+    end
+
+    def cast_boolean
+      return true if TRUE_VALUES.include? value
+      return false if FALSE_VALUES.include? value
+      raise TypeError
+    end
+
+    def cast_integer
+      return value.to_i if value.is_a?(Numeric)
+
+      if value.is_a?(String)
+        if value =~ /^0x[0-9a-f]+$/i
+          value.to_i(16)
+        elsif value =~ /^0[0-7]+$/ #this is stupid, "14" will be translated to 12 :(
```

Comment:
hehe, I wrote this comment and forgot to remove it.
wrote this comment because I misread the regexp. "14" will be 14, "014" will be 12 (the leading zero says "this is octal")
I'll remove the comment :smile: